### PR TITLE
show headers during GithubException

### PR DESCRIPTION
**Pull request by IanButterworth**
Trying to check for information we currently ignore that could investigate https://github.com/JuliaRegistries/TagBot/issues/298

### CODE REVIEW

**Comment by IanButterworth** [comment]
My python knowledge is getting in the way

https://github.com/PyGithub/PyGithub/blob/673974918882d68826659daa5435e833cd7148c2/github/GithubException.py#L74C9-L74C16

```
>           logger.info(e.headers())
E           AttributeError: 'GithubException' object has no attribute 'headers'
```

**Comment by DilumAluthge** [comment]
Why version of pygithub are we using in our CI here?

**Comment by IanButterworth** [comment]
1.54.1 which doesn't store headers
https://github.com/PyGithub/PyGithub/blob/34d097ce473601624722b90fc5d0396011dd3acb/github/GithubException.py

**Comment by DilumAluthge** [comment]
😭

**Comment by DilumAluthge** [comment]
What's the first version of pygithub that stores the headers?

File: pyproject.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ license = "MIT"
 python = "^3.8"
 Flask = "^1.1.2"
 Jinja2 = "^2.11.2"
-PyGithub = "^1.51"
+PyGithub = "1.58.2"
```
Comment (DilumAluthge) [inline pyproject.toml:12]: Is it intentional that you dropped the caret?

Comment (IanButterworth) [reply pyproject.toml:12]: yes. 1.59 had breaking changes

**Comment by IanButterworth** [comment]
woo. Now we just need someone with a failing TagBot to try this out

**Comment by IanButterworth** [comment]
Or maybe we just release this as a patch and see in the wild

**Comment by IanButterworth** [comment]
Let's do it. It's green and we can revert in a patch if needed

**Pull request merged**